metaserver/src/main/scala/scala/meta/languageserver/search/InMemorySymbolIndex.scala

```diff
@@ -0,0 +1,226 @@
+package scala.meta.languageserver.search
+
+import java.net.URI
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.concurrent.ConcurrentHashMap
+import scala.meta.languageserver.Buffers
+import scala.meta.languageserver.Effects
+import scala.meta.languageserver.ScalametaEnrichments._
+import scala.meta.languageserver.ServerConfig
+import scala.meta.languageserver.compiler.CompilerConfig
+import scala.meta.languageserver.mtags.Mtags
+import scala.meta.languageserver.ScalametaLanguageServer.cacheDirectory
+import scala.meta.languageserver.storage.LevelDBMap
+import scala.meta.languageserver.{index => i}
+import `scala`.meta.languageserver.index.Position
+import `scala`.meta.languageserver.index.SymbolData
+import com.typesafe.scalalogging.LazyLogging
+import langserver.{types => l}
+import langserver.core.Notifications
+import langserver.messages.DefinitionResult
+import langserver.messages.ReferencesResult
+import langserver.messages.DocumentSymbolResult
+import org.langmeta.inputs.Input
+import org.langmeta.internal.io.FileIO
+import org.langmeta.internal.semanticdb.schema.Database
+import org.langmeta.internal.semanticdb.schema.Document
+import org.langmeta.internal.semanticdb.schema.ResolvedName
+import org.langmeta.internal.semanticdb.{schema => s}
+import org.langmeta.io.AbsolutePath
+import org.langmeta.io.RelativePath
+import org.langmeta.languageserver.InputEnrichments._
+import org.langmeta.semanticdb.Signature
+import org.langmeta.semanticdb.Symbol
+
+class InMemorySymbolIndex(
+    val symbolIndexer: SymbolIndexer,
+    val documentIndex: DocumentIndex,
+    cwd: AbsolutePath,
+    notifications: Notifications,
+    buffers: Buffers,
+    serverConfig: ServerConfig,
+) extends SymbolIndex with LazyLogging {
+  private val indexedJars: ConcurrentHashMap[AbsolutePath, Unit] =
+    new ConcurrentHashMap[AbsolutePath, Unit]()
+
+  /** Returns a ResolvedName at the given location */
+  def resolveName(
+      path: AbsolutePath,
+      line: Int,
+      column: Int
+  ): Option[ResolvedName] = {
+    logger.info(s"resolveName at $path:$line:$column")
+    for {
+      document <- documentIndex.getDocument(path.toNIO.toUri)
+      _ = logger.info(s"Found document for $path")
+      _ <- isFreshSemanticdb(path, document)
```

Comment:
This is odd.  The naming convension says it should return a Boolean, the doc comment says it returns a bool, yet it returns Option[Unit] and performs side effects.